project/LIS2DH12.c

```diff
@@ -63,14 +63,15 @@ typedef struct {
 /*******************************************************************************
 															VARIABLES AND CONSTANTS
 *******************************************************************************/
+lis2dh12_instance_t accel_inst = {0}; // Instance of the LIS2DH12
 
 #define SPI_INSTANCE  0 //SPI instance index
 const nrf_drv_spi_t spi = NRF_DRV_SPI_INSTANCE(SPI_INSTANCE); //SPI instance
 volatile bool spi_xfer_done; //Flag used to indicate that SPI instance completed the transfer
 
 #define SPI_BUFFER_LENGTH (192)
 uint8_t m_tx_buf[SPI_BUFFER_LENGTH]; //Tx buffer
-uint8_t m_rx_buf[SPI_BUFFER_LENGTH]; //Rx buffer
+uint8_t m_rx_buf[SPI_BUFFER_LENGTH + 1]; //Rx buffer
```

Comment:
Why is this necessary?

Reply:
This is necessary because if we are receiving 192 bytes, then the rx buffer needs to be at least 193 bytes, because received data starts at index 1 and not index 0. If we added 10% or if we just made it 256 instead, this wouldn't be necessary, but then we would have to adjust the read function. I am good either way